led_v3/code/MCAL/MGPIO/mgpio_Program.c

```diff
@@ -0,0 +1,308 @@
+ /**************************************************************
+ *  AUTHOR 						: Bassel Yasser
+ *  Version 					: V0.0.0
+ *  Date							: 13 - 6 - 2023
+ *  Description 			: MGPIO_Interface.h --> Function Prototypes
+ *  Module Features		:
+ *  											01- MGPIO_u8Init()
+ *  											02- MGPIO_u8SetPinData()
+													03-	MGPIO_u8GetPinData()
+													04-	MGPIO_u8IRQEnable()
+													05-	MGPIO_vidIRQDisable()
+													06-	MGPIO_u8SetCallBack()
+ *****************************************************************************/
+
+/********************************************************************************************************************
+*												Standard Types LIB
+********************************************************************************************************************/
+//#include "LSTD_TYPES.h"
+#include "LBIT_MATH.h"
+/********************************************************************************************************************
+*												Peripheral Files
+********************************************************************************************************************/
+#include "mgpio_Private.h"
+#include "mgpio_Interface.h"
+#include "mgpio_Config.h"
+/********************************************************************************************************************
+*												Global Variables
+********************************************************************************************************************/
+
+static ptr_func_t gl_a_ptr_func[6] = {NULL,NULL,NULL,NULL,NULL,NULL};
+
+/********************************************************************************************************************
+ *  										Function Implementation
+*******************************************************************************************************************/
+uint8_ MGPIO_u8Init(st_gpio_cfg_t* st_gpio_cfg)
+{
+	uint8_ loc_u8_error_status  = MGPIO_SUCCESS;
+
+	uint8_ loc_u8_PortNum;
+	uint8_ loc_u8_PinNum;
+
+	if (st_gpio_cfg != NULL)
+	{
+		if (st_gpio_cfg->enu_pin 						< MGPIO_PIN_INVALID			&&
+				st_gpio_cfg->enu_gpio_mode 			< MGPIO_MODE_INVALID		&&
+				st_gpio_cfg->enu_pin_dir_mode 	< MGPIO_DIR_INVALID			&&
+				((st_gpio_cfg->un_gpio_conf.enu_gpio_amp_mode 	< MGPIO_MAMP_INVALID) || (st_gpio_cfg->un_gpio_conf.u8_input_pull_type < MGPIO_PULL_TYPE_MAX))    )
+		{
+			loc_u8_PortNum 	= st_gpio_cfg->enu_pin / 8;
+			loc_u8_PinNum		= st_gpio_cfg->enu_pin % 8;
+			
+			SET_BIT(RCGCGPIO_REG, loc_u8_PortNum );
+			
+			switch (st_gpio_cfg->enu_gpio_mode)
+			{
+				case MGPIO_MODE_DIGITAL:
+							switch (st_gpio_cfg->enu_pin_dir_mode)
+							{
+								case MGPIO_DIR_INPUT	:	CLR_BIT(GPIODIR(loc_u8_PortNum), loc_u8_PinNum );
+																				switch (st_gpio_cfg->un_gpio_conf.u8_input_pull_type)
```

Comment:
Switch should have only one break

Reply:
Hello, Eng. Hala
I hope u r doing well

We couldn't understand how we can use switch case with one break, if we did it would result in a logical error
And we looked at this rule on the internet, there is no such rule, if you have a link or any source explaining this rule, let us understand it more.

thank you,

Reply:
I don't know the exact version which mentions this rule so you have two solutions to avoid doing switch and to be compliant with  this MISRA rule:

1. The easy way to replace the switch case with if condition.
2. The hard way is to use Fall-through switch cases.

Reply:
Ok, it resolved , MGPIO in all task version is modified